master/static/migrations/20240201101714_generic-task-migrations.tx.up.sql

```diff
@@ -0,0 +1,55 @@
+ALTER TYPE task_type RENAME TO _task_type;
+
+CREATE TYPE task_type AS ENUM (
+  'TRIAL',
+  'NOTEBOOK',
+  'SHELL',
+  'COMMAND',
+  'TENSORBOARD',
+  'CHECKPOINT_GC',
+  'GENERIC'
```

Comment:
@NicholasBlaskey you were looking at [a way to optimize add enum values](https://hpe-aiatscale.atlassian.net/browse/DET-10099)
is that something we could easily add here? the `tasks` table at certain customers might be quite large but they're running newer postgres.

Reply:
Maybe like something this could work?
```
DO $$ 
BEGIN
    IF current_setting('server_version_num')::int > 120000 THEN
       ALTER TYPE enum ADD value 'value';
    ELSE
      -- DO hack to add enums that we do
    END IF;
END $$;
```

I would double check that 12 is actually the version that starts supporting this

Reply:
If you rebase the performance test stuff is landed

So if you check the approve performance test in circleci, under ```e2e-longrunning``` there will be a step printing the migration time. You can use this before adding the VERSION polyfill to verify this will cause customer impact, and then after the VERSION polyfill to show that the migration completes with little impact.

Reply:
Seems like `ALTER TYPE enum ...` only supports adding an enum and not dropping one. So we still have to use the old way for the down migration